File: JAVA/Juggler Sequence.java
class Solution {
    static List<Long> jugglerSequence(Long n) {
        List<Long> arr= new ArrayList<Long>();
        arr.add(n);
        while(n>1){
            if(n%2==0){
                n =(long) Math.pow(n,(float)1/2);
                arr.add(n);
            }
            else{
                n= (long)Math.pow(n,(float)3/2);
                arr.add(n);
            }
        }
        return arr;
        
    }
}

File: C++/Juggler Sequence.cpp
class Solution {
  public:
    vector<long long> jugglerSequence(long long n)
{
    vector<long long> ans;
    ans.push_back(n);
    while (n != 1)
    {
        if (n % 2 == 0)
        {
            n = sqrt(n);
        }
        else
        {
            n = sqrt(n) * sqrt(n) * sqrt(n);
        }
        ans.push_back(n);
    }
    return ans;
}
};
